src/Bucket/Bucket.jl

```diff
@@ -0,0 +1,419 @@
+    atmosphere::ATM,
+    radiation::RAD,
+) where {FT, PSE, ATM, RAD}
+    args = (parameters, atmosphere, radiation, domain)
+    BucketModel{FT, typeof.(args)...}(args...)
+end
+
+prognostic_types(::BucketModel{FT}) where {FT} = (FT, FT, FT, FT)
+prognostic_vars(::BucketModel) = (:W, :T_sfc, :Ws, :S)
+auxiliary_types(::BucketModel{FT}) where {FT} = (FT, FT, FT, FT, FT)
+auxiliary_vars(::BucketModel) = (:q_sfc, :E, :LHF, :SHF, :R_n)
+
+"""
+    surface_fluxes( T_sfc::FT,
+                    q_sfc::FT,
+                    S::FT
+                    t::FT,
+                    parameters::P,
+                    atmos::PA,
+                    radiation::PR,
+                    ) where {FT <: AbstractFloat, P <: BucketModelParameters{FT},  PA <: PrescribedAtmosphere{FT}, PR <: PrescribedRadiation{FT}}
+
+Computes the surface flux terms at the ground for a standalone simulation:
+net radiation,  SHF,  LHF,
+as well as the water vapor flux (in units of m^3/m^2/s of water).
+Positive fluxes indicate flow from the ground to the atmosphere.
+
+It solves for these given atmospheric conditions, stored in `atmos`,
+ downwelling shortwave and longwave radiation (in `radiation`),
+model parameters, and the surface temperature and surface specific
+humidity.
+
+Currently, we only support soil covered surfaces.
+"""
+function surface_fluxes(
+    T_sfc::FT,
+    q_sfc::FT,
+    S::FT,
+    t::FT,
+    parameters::P,
+    atmos::PA,
+    radiation::PR,
+) where {
+    FT <: AbstractFloat,
+    P <: BucketModelParameters{FT},
+    PA <: PrescribedAtmosphere{FT},
+    PR <: PrescribedRadiation{FT},
+}
+    @unpack ρ_atmos, T_atmos, u_atmos, q_atmos, h_atmos, ρ_sfc = atmos
+    @unpack α_soil, α_snow, z_0m, z_0b, S_c, earth_param_set = parameters
+    @unpack LW_d, SW_d = radiation
+    _σ = Stefan()
+    _ρ_liq = ρ_cloud_liq(earth_param_set)
+
+    # call surface fluxes for E, SHF, LHF
+    ts_sfc = Thermodynamics.PhaseEquil_ρTq(earth_param_set, ρ_sfc, T_sfc, q_sfc)
+    ts_in = Thermodynamics.PhaseEquil_ρTq(
+        earth_param_set,
+        ρ_atmos(t),
+        T_atmos(t),
+        q_atmos(t),
+    )
+
+    state_sfc = SurfaceFluxes.SurfaceValues(FT(0), SVector{2, FT}(0, 0), ts_sfc)
+    state_in = SurfaceFluxes.InteriorValues(
+        h_atmos,
+        SVector{2, FT}(u_atmos(t), 0),
+        ts_in,
+    )
+
+    # State containers
+    sc = SurfaceFluxes.ValuesOnly{FT}(;
+        state_in,
+        state_sfc,
+        z0m = z_0m,
+        z0b = z_0b,
+    )
+
+    conditions = SurfaceFluxes.surface_conditions(
+        earth_param_set,
+        sc,
+        SurfaceFluxes.UniversalFunctions.Businger(),
+    )
+
+    α = surface_albedo(α_soil, α_snow, S, S_c)
+    R_n = -((FT(1) - α) * SW_d(t) + LW_d(t) - _σ * T_sfc^FT(4.0))
+    # Land needs a volume flux of water, not mass flux
+    # If this evaporation is positive -> towards atmos
+    E = SurfaceFluxes.evaporation(sc, earth_param_set, conditions.Ch) / _ρ_liq
```

Comment:
I will add `E` to the output of SF so you won't need to call it here in the future. for now this is fine 
@akshaysridhar what do you think ?